### PR TITLE
Add google benchmark to `cuDF`

### PR DESCRIPTION
**Pull request by hummingtree**
Add the google benchmark ([https://github.com/google/benchmark](url)) framework to `cuDF`.

This PR depends on PR#1891 "Add equality operator for GDF_STRING_CATEGORY and refactor binops" and PR#1987 "Allow user defined functions in the form of ptx code to be passed to binops". 

The above two PRs should be merged first.

I will make another PR that does not depends these two PRs. My apology.

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -188,6 +191,23 @@ if(BUILD_TESTS)
...
 
+###################################################################################################
+# - add google benchmark --------------------------------------------------------------------------
+
+if(BUILD_BENCHMARKS)
+  
+  include(ConfigureGoogleBenchmark)
+  
+  if(GBENCH_FOUND)
+    message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GTEST_ROOT}")
```
Comment (kkraus14) [inline cpp/CMakeLists.txt]:
```suggestion
    message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GBENCH_ROOT}")
```

File: cpp/benchmarks/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+###################################################################################################
+# - compiler function -----------------------------------------------------------------------------
+
+function(ConfigureBench CMAKE_BENCH_NAME CMAKE_BENCH_SRC)
+    add_executable(${CMAKE_BENCH_NAME}
+                   ${CMAKE_BENCH_SRC})
+                   # "${CMAKE_CURRENT_SOURCE_DIR}/utilities/cudf_test_utils.cu")
+    set_target_properties(${CMAKE_BENCH_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+    target_link_libraries(${CMAKE_BENCH_NAME} gmock gtest gmock_main gtest_main benchmark pthread cudf)
```
[the remark this answers is not in the document]
Comment (hummingtree) [reply cpp/benchmarks/CMakeLists.txt]: gbench depends on gtest but I am not sure if they are all necessary.

**Comment by harrism** [comment]
Why did you close this?

**Comment by hummingtree** [comment]
I will open another PR that does not depend on the two PRs mentioned here.

**Pull request closed without merging**